### DISCUSSION TITLE
How does LazyVim get such well formatted and extensive cmp documentation for their completion options?

### DISCUSSION THREAD

**Discussion by timproved** [LazyVim/LazyVim#3519] [Q&A] [answered]
I have noticed LazyVims documentation on completion items seems to be more extensive compared to other distrib., how does LazyVim mange this? 
Is this an extra plugin or is this done in the util.cmp.lua?

**Comment by folke** [comment]
You mean for completion in your personal config?
use lazydev.nvim

**Reply by timproved** [reply]
Sorry if I have caused confusion, I should've provided screenshots from the get go.
This is want I mean:
![screenshot-2024-06-07-172348](https://github.com/LazyVim/LazyVim/assets/48419301/88177785-1278-4e67-9ad2-16d1b5502d34)
Here on the left the documentation to PageRequest.of is well formatted and seems to resolve and render the hyperlinks contents.
Whereas here:
![screenshot-2024-06-07-171504](https://github.com/LazyVim/LazyVim/assets/48419301/3a18250b-43b4-43f4-90fb-ef2b8bccc9ba)
The Hyperlink isn't being resolved and the formatting isn't present.

**Reply by dpetka2001** [reply, marked as answer]
This must come from `noice` configuration [here](https://github.com/LazyVim/LazyVim/blob/913902ba362603a4620f8981fdedac7f53816b3a/lua/lazyvim/plugins/ui.lua#L274-L280) I believe.

**Reply by timproved** [reply]
You are right, it  absolutely does!!!
Amazing thank you @dpetka2001 and @folke 
The result with noice.nvim, now matches the Appearance of LazyVim (minus the better formatting):
![screenshot-2024-06-07-174135](https://github.com/LazyVim/LazyVim/assets/48419301/3b7265b7-4fff-436d-8e9e-a282e3219561)

**Comment by dpetka2001** [comment]
The only thing I see in `util.cmp` adding to the docs is missing documentation for snippets. The rest should be default `nvim-cmp`. About the formatting you can see this [code snippet](https://github.com/LazyVim/LazyVim/blob/c1cb4467ec78b309784f948a47883ab11a2eecd9/lua/lazyvim/plugins/coding.lua#L51-L59), which is responsible about how the auto-completion menu will look like.
